javasphinx/htmlrst.py

```diff
@@ -395,8 +397,8 @@ def _preprocess(self, s_html):
     # ---- Conversion entry point ----
 
     def convert(self, s_html):
-        if not isinstance(s_html, unicode):
-            s_html = unicode(s_html, 'utf8')
+        if not isinstance(s_html, str):
+            s_html = str(s_html, 'utf8')
```

Comment:
The meaning of `str` here will vary between Python 2.x and 3.x. I think we want `from builtins import str` at the top of this file and this will be good. I'd also recommend removing the `basestring` import and updating the `_process` method to check `isinstance(node, str)` instead of `isinstance(node, basestring)`.